manim/animation/movement.py

```diff
@@ -44,6 +44,17 @@ class Homotopy(Animation):
         Keyword arguments propagated to :meth:`.Mobject.apply_function`.
     kwargs
         Further keyword arguments passed to the parent class.
+
+    Examples
+    --------
+    .. manim:: HomotopyExample
+
+        class HomotopyExample(Scene):
+            def construct(self):
+                dot = Dot()
+                homotopy = lambda x,y,z,t:[x + t, y + t**2, z]
+
+                self.play(Homotopy(homotopy, dot))
```

Comment:
Currently it look just like shifting the dot. How about something like this waving square:
```suggestion
                square = Square()
                homotopy = lambda x, y, z, t: (x, y + 0.2 * np.sin(x + 10 * t), z)
                self.play(Homotopy(homotopy, square))
```

https://user-images.githubusercontent.com/43041139/192983555-1a602921-51fd-4b66-97b9-9ede4757fcda.mp4